### PR TITLE
export integrated cookie functionality

### PR DESCRIPTION
**Pull request by Uzlopak**
This way we can use it in fastify/secure-session and co instead of the cookie package.

#### Checklist

- [ ] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Review by gurgunday — APPROVED** [review]
Great, thanks

File: cookie.js [hunk 1]
```diff
@@ -223,3 +215,13 @@ function tryDecode (str, decode) {
...
+
+/**
+ * Module exports.
+ * @public
+ */
+
+module.exports = {
+  parse,
+  serialize
+}
```
Comment (mcollina) [inline cookie.js:227]: please restore the original setup, that works with ESM while this does not.

**Review by mcollina — CHANGES_REQUESTED** [review]

File: plugin.js [hunk 2]
```diff
@@ -214,6 +214,8 @@ module.exports = fastifyCookie
 module.exports.default = fastifyCookie // supersedes fastifyCookie.default = fastifyCookie
 module.exports.fastifyCookie = fastifyCookie // supersedes fastifyCookie.fastifyCookie = fastifyCookie
 
+module.exports.cookie = cookie
```
Comment (mcollina) [inline plugin.js:217]: Can you change our test to load cookie from the top level module then?

**Review by mcollina — CHANGES_REQUESTED** [review]

File: cookie.js (hunk 1 above)
Comment (gurgunday) [reply cookie.js:227]: Yeah but it’s a named export from the main file

So you ideally import everything from there

File: plugin.js (hunk 2 above)
Comment (Uzlopak) [reply plugin.js]: idk

**Comment by Uzlopak** [comment]
@mcollina PTAL

**Comment by mcollina** [comment]
Plase require the entrypoint file in https://github.com/fastify/fastify-cookie/blob/efea6715583149fdc7a591ca831047a84e3b80ce/test/cookie-module.test.js#L6.

**Review by jsumners — COMMENTED** [review]
If the other module is going to use `@fastify/cookie` to manage cookies, why do we need to export the underlying tooling?

**Comment by Uzlopak** [comment]
@mcollina 
PTAL

**Review by gurgunday — APPROVED** [review]

**Review by mcollina — CHANGES_REQUESTED** [review]
Can you document it in the README?

File: README.md [hunk 3]
```diff
@@ -11,6 +11,8 @@ This plugin's cookie parsing works via Fastify's `onRequest` hook. Therefore,
 you should register it prior to any other `onRequest` hooks that will depend
 upon this plugin's actions.
 
+It is also possible to import the low-level cookie parsing and serialization functions.
```
Comment (Fdawgs) [inline README.md:14]: Maybe add a link here to the header section regarding the import?

**Review by Fdawgs — APPROVED** [review]

**Comment by Uzlopak** [comment]
@mcollina PTAL

@gurgunday 
Thank you for your work.

**Review by mcollina — APPROVED** [review]
lgtm

**Pull request merged**